index.js

```diff
@@ -2,23 +2,29 @@
 
 const Promise = require('bluebird');
 const readInput = Promise.promisify(require('read'));
-const argv = require('yargs').argv;
+const argv = require('yargs').array('domain').argv;
 const xtend = require('xtend');
 const getCertificate = require('./lib.js');
 
-const read = (options) => () => {
+const read = (options) => (forceReadInput = false) => {
     const validator = options.validator ? options.validator : v => v && v.length > 0;
-    const argument = typeof argv[options.argument] === 'string' ? argv[options.argument] : undefined;
-    const resultPromise = argument ? Promise.resolve(argument) : readInput(options);
-    return resultPromise.then(value => {
-        if (validator(value)) {
+    const processValue = options.processValue ? options.processValue : v => v;
```

Comment:
Is this needed? My understanding is that yargs handles this automatically by specifying that domain is expected to be an array. If only one domain is specified, an array of 1 element is returned.